circuit/src/commonMain/kotlin/com/slack/circuit/CircuitContent.kt

```diff
@@ -72,7 +72,10 @@ internal fun CircuitContent(
     remember(screen, context) {
       circuitConfig.eventListenerFactory?.create(screen, context) ?: EventListener.NONE
     }
-  DisposableEffect(screen, context) { onDispose { eventListener.dispose() } }
+  DisposableEffect(screen, context) {
+    eventListener.start()
```

Comment:
I think this should be moved up to the remember block that creates the listener

Reply:
There wouldn't be a functional difference, but it is more correct to me. This block is for disposal, whereas the above block is initialization of the listener and thus its starting point

Reply:
> ...it is more correct to me

Hmm I don't think I agree with this. I'd argue that this is precisely what `DisposableEffect` is for. The official documentation has an [example](https://developer.android.com/jetpack/compose/side-effects#disposableeffect) that does something similar. Further, I think the method docs agree with me on this:

> A side effect of composition that must run for any new unique value of key1 or key2 and must be reversed or cleaned up if key1 or key2 changes, or if the DisposableEffect leaves the composition.

Reply:
I don't think that example is necessarily analogous to this case, but my LOGAF is low on this so if you feel strongly we can do it this way 👍